pystac_monty/sources/idu.py

```diff
@@ -0,0 +1,231 @@
+
+        item = Item(
+            id=f'{STAC_EVENT_ID_PREFIX}{data["event_id"]}',
+            geometry=geometry,
+            bbox=bbox,
+            datetime=startdate,
+            properties={
+                "title": data["event_name"],
+                "description": description,
+                "start_datetime": startdate.isoformat(),
+                "end_datetime": enddate.isoformat(),
+                "location": data["locations_name"],
+                "sources": data["sources"]
+            }
+        )
+
+        item.set_collection(self.get_event_collection())
+        item.properties["roles"] = ["source", "event"]
+
+        item.add_asset(
+            "report",
+            Asset(
+                href=data["source_url"],
+                media_type="application/pdf",
+                title="Report"
+            )
+        )
+        item.add_link(Link("via", self.data.get_source_url(), "application/json", "IDU Event Data"))
+
+        hazard_tuple = (
+            data["category"],
+            data["subcategory"],
+            data["type"],
+            data["subtype"]
+        )
+
+        MontyExtension.add_to(item)
+        monty = MontyExtension.ext(item)
+        monty.episode_number = episode_number
+        monty.country_codes = [data["iso3"]]
+        monty.hazard_codes = self.map_idu_to_hazard_codes(hazard=hazard_tuple)
+        monty.compute_and_set_correlation_id(hazard_profiles=self.hazard_profiles)
+
+        return item
+
+    def map_idu_to_hazard_codes(self, hazard: tuple):
+        """Map IDU hazards to UNDRR-ISC 2020 Hazard Codes"""
+        hazard = tuple((item.lower() if item else item for item in hazard))
+        hazard_mapping = {
+            ('geophysical', 'geophysical', 'earthquake', 'earthquake'): ["GH0001", "GH0004"],
+            ('geophysical', 'geophysical', 'mass movement', 'dry mass movement'): ["Dry Mass Movement"],
+            ('geophysical', 'geophysical', 'mass movement', 'sinkhole'): ["GH0026"],
+            ('geophysical', 'geophysical', 'volcanic activity', 'volcanic activity'): ["GH0020"],
+            ('mixed disasters', 'mixed disasters', 'mixed disasters', 'mixed disasters'): "Mixed Disaster",
+            ('weather related', 'climatological', 'drought', 'drought'): ["MH0035"],
+            ('weather related', 'climatological', 'erosion', 'erosion'): ["EN0019"],
+            ('weather related', 'climatological', 'sea level rise', 'sea level rise'): ["EN0023"],
+            ('weather related', 'climatological', 'wildfire', 'wildfire'): ["EN0013"],
+            ('weather related', 'hydrological', 'flood', 'dam release flood'): ["TL0009"],
+            ('weather related', 'hydrological', 'flood', 'flood'): ["FL"],
+            ('weather related', 'hydrological', 'mass movement', 'avalanche'): ["MH0050"],
+            ('weather related', 'hydrological', 'mass movement', 'landslide/wet mass movement'): ["Landslide/Wet Mass Movement"],
```

Comment:
May be we need to set suitable hazard code for landslide.

Reply:
Done

Reply:
@emmanuelmathot  can you suggest me the hazard code for `Mixed Disaster` and `Salinisation`. We are facing problem in creating stac items for this hazards.